src/contentScript/stateController.test.ts

```diff
@@ -70,4 +75,55 @@ describe("pageState", () => {
       asyncState: { isFetching: true, currentData: null, data: "foo" },
     });
   });
+
+  describe("mod variable policy", () => {
+    it("stores variable in session storage", async () => {
+      const listener = jest.fn();
+
+      document.addEventListener(STATE_CHANGE_JS_EVENT_TYPE, listener);
+
+      const modComponentRef = modComponentRefFactory();
+
+      await expect(browser.storage.session.get(null)).resolves.toStrictEqual(
+        {},
+      );
+
+      registerModVariables(modComponentRef.modId, {
+        schema: {
+          type: "object",
+          properties: {
+            foo: {
+              type: "object",
+              "x-sync-policy": "session",
+              // Cast required because types don't support custom `x-` variables yet
+            } as JSONSchema7Definition,
+          },
+          required: ["foo"],
+        },
+      });
+
+      await setState({
+        namespace: StateNamespaces.MOD,
+        data: { foo: { bar: "baz" }, quox: 42 },
+        mergeStrategy: MergeStrategies.REPLACE,
+        modComponentRef,
+      });
+
+      // The storage fake doesn't emit events
+      expect(listener).toHaveBeenCalledTimes(0);
+
+      const values = await browser.storage.session.get(null);
+
+      // Ensure values are segmented correctly in storage
+      expect(JSON.stringify(values)).not.toContain("quox");
+      expect(JSON.stringify(values)).toContain("bar");
```

Comment:
NIT: Why not be more explicit (and document the behavior better:

```suggestion
       expect(values).toEqual({
        [getSessionStorageKey(modComponentRef.modId)]: { foo: { bar: "baz" } },
      });
```

Reply:
I shied away from it because ideally the key structure would be opaque. But agree the clarity in test code would be more beneficial. I added a comment on the method export